rlclientlib/model_mgmt/restapi_data_transport.cc

```diff
@@ -38,50 +41,62 @@ namespace reinforcement_learning { namespace model_management {
    * x-ms-version = 2017-04-17
    */
 
-  int restapi_data_transport::get_data_info(::utility::datetime& last_modified, ::utility::size64_t& sz, api_status* status) {
+  int restapi_data_transport::get_data_info(::utility::datetime& last_modified, ::utility::size64_t& sz, api_status* status, http_request request) {
 
-    // Build request URI and start the request.
-    auto request_task = _httpcli->request(methods::HEAD)
-      // Handle response headers arriving.
-      .then([&](http_response response) {
-      if ( response.status_code() != 200 )
-        RETURN_ERROR_ARG(_trace, status, http_bad_status_code, _httpcli->get_url());
+      // Get request URI and start the request.
+      request.headers() = _header;
+      // Build request URI and start the request.
+      auto request_task = _httpcli->request(request).then([&](http_response response) {
+          if ( response.status_code() != 200 )
+              RETURN_ERROR_ARG(_trace, status, http_bad_status_code, _httpcli->get_url());
 
-      const auto iter = response.headers().find(U("Last-Modified"));
-      if ( iter == response.headers().end() )
-        RETURN_ERROR_ARG(_trace, status, last_modified_not_found, _httpcli->get_url());
+          const auto iter = response.headers().find(U("Last-Modified"));
+          if ( iter == response.headers().end() )
+              RETURN_ERROR_ARG(_trace, status, last_modified_not_found, _httpcli->get_url());
 
-      last_modified = ::utility::datetime::from_string(iter->second);
-      if( last_modified.to_interval() == 0)
-        RETURN_ERROR_ARG(_trace, status, last_modified_invalid, _httpcli->get_url());
+          last_modified = ::utility::datetime::from_string(iter->second);
+          if( last_modified.to_interval() == 0)

```

Comment:
Would be a case the last_modified to be negative?

Reply:
as per my understanding to_interval returns in 100's of nano seconds. So if the value is 0 it means date is invalid otherwise it will be in some positive value